docs/RFC/draft-paragonie-paseto-rfc-00.txt

```diff
@@ -0,0 +1,1680 @@
+
+                  cccccccc  cccccccc  cccccccc  cccccccc
+                  kkkkkkkk  kkkkkkkk  kkkkkkkk  kkkkkkkk
+                  kkkkkkkk  kkkkkkkk  kkkkkkkk  kkkkkkkk
+                  nnnnnnnn  nnnnnnnn  nnnnnnnn  nnnnnnnn
+
+                 HChaCha20 State: c=constant k=key n=nonce
+
+   After initialization, proceed through the ChaCha rounds as usual.
+
+   Once the 20 ChaCha rounds have been completed, the first 128 bits and
+   last 128 bits of the keystream (both little-endian) are concatenated,
+   and this 256-bit subkey is returned.
+
+7.2.1.  Test Vector for the HChaCha20 Block Function
+
+   o  Key = 00:01:02:03:04:05:06:07:08:09:0a:0b:0c:0d:0e:0f:10:11:12:13:
+      14:15:16:17:18:19:1a:1b:1c:1d:1e:1f.  The key is a sequence of
+      octets with no particular structure before we copy it into the
+      HChaCha state.
+
+   o  Nonce = (00:00:00:09:00:00:00:4a:00:00:00:00:31:41:59:27)
+
+   After setting up the HChaCha state, it looks like this:
+
+
+
+
+
+
+
+
+Arciszewski & Haussmann Expires October 21, 2018               [Page 18]
+
+Internet-Draft                                                April 2018
+
+
+                    61707865 3320646e 79622d32 6b206574
+                    03020100 07060504 0b0a0908 0f0e0d0c
+                    13121110 17161514 1b1a1918 1f1e1d1c
+                    09000000 4a000000 00000000 27594131
+
+                     ChaCha state with the key setup.
+
+   After running 20 rounds (10 column rounds interleaved with 10
+   "diagonal rounds"), the HChaCha state looks like this:
+
+                    82413b42 27b27bfe d30e4250 8a877d73
+                    4864a70a f3cd5479 37cd6a84 ad583c7b
+                    8355e377 127ce783 2d6a07e0 e5d06cbc
+                    a0f9e4d5 8a74a853 c12ec413 26d3ecdc
+
+                       HChaCha state after 20 rounds
+
+   HChaCha20 will then return only the first and last rows, resulting in
+   the following 256-bit key:
+
+                    82413b4 227b27bfe d30e4250 8a877d73
+                    a0f9e4d 58a74a853 c12ec413 26d3ecdc
+
+                        Resultant HChaCha20 subkey
+
+8.  Intended Use-Cases for PASETO
+
+   Like JWTs, PASETOs are intended to be single-use tokens, as there is
+   no build-in mechanism to prevent replay attacks within the token
+   lifetime.
+
+   o  *local* tokens are intended for tamper-resistant encrypted cookies
+      or HTTP request parameters.  A resonable example would be long-
+      term authentication cookies which re-establish a new session
+      cookie if a user checked the "remember me on this computer" box
+      when authenticating.  To accomplish this, the server would look
+      use the "jti" claim in a database lookup to find the appropriate
+      user to associate this session with.  After each new browsing
+      session, the "jti" would be rotated in the database and a fresh
+      cookie would be stored in tbe browser.
```

Comment:
the

Also, I'm surprised to see you promoting this use case, having railed against it.